File: license-server/src/main/java/de/netzkronehd/license/model/LicenseKeyModel.java
package de.netzkronehd.license.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.*;

import java.util.Objects;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
public class LicenseKeyModel {

    @Id
    private String owner;
    @Column(columnDefinition = "TEXT")
    private String privateKey;
    @Column(columnDefinition = "TEXT")
    private String publicKey;

    @Override
    public final boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof LicenseKeyModel that)) return false;
        return Objects.equals(owner, that.owner) && Objects.equals(privateKey, that.privateKey) && Objects.equals(publicKey, that.publicKey);
    }

    @Override
    public int hashCode() {
        int result = Objects.hashCode(owner);
        result = 31 * result + Objects.hashCode(privateKey);
        result = 31 * result + Objects.hashCode(publicKey);
        return result;
    }
}

File: license-server-rs/src/model/license_key_model.rs
use std::fmt::{Display, Formatter};

pub struct LicenseKeyModel {
    owner: String,
    private_key: String,
    public_key: String,
}

impl LicenseKeyModel {
    pub fn owner(&self) -> &str {
        &self.owner
    }

    pub fn private_key(&self) -> &str {
        &self.private_key
    }

    pub fn public_key(&self) -> &str {
        &self.public_key
    }

}

impl PartialEq for LicenseKeyModel {
    fn eq(&self, other: &Self) -> bool {
        self.owner == other.owner
            && self.private_key == other.private_key
            && self.public_key == other.public_key
    }
}

impl Display for LicenseKeyModel {
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        write!(
            f,
            "LicenseKeyModel {{ owner: {:?}, public_key: {:?} }}",
            self.owner, self.public_key
        )
    }
}
